api/v1alpha6/types.go

```diff
@@ -117,11 +117,9 @@ type PortOpts struct {
 	FixedIPs  []FixedIP `json:"fixedIPs,omitempty"`
 	TenantID  string    `json:"tenantId,omitempty"`
 	ProjectID string    `json:"projectId,omitempty"`
-	// The uuids of the security groups to assign to the instance
-	SecurityGroups *[]string `json:"securityGroups,omitempty"`
-	// The names, uuids, filters or any combination these of the security groups to assign to the instance
-	SecurityGroupFilters []SecurityGroupParam `json:"securityGroupFilters,omitempty"`
-	AllowedAddressPairs  []AddressPair        `json:"allowedAddressPairs,omitempty"`
+	// The names of the security groups to assign to the port
+	SecurityGroups      *[]SecurityGroupParam `json:"securityGroups,omitempty"`
```

Comment:
Please check this carefully.
I have copied it from https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1257 but I'm unsure if it should be `*[]SecurityGroupParam` or `[]SecurityGroupParam`. Note that the `SecurityGroupFilters` used to be `[]SecurityGroupParam` but now `SecurityGroups` is `*[]SecurityGroupParam`.